### DISCUSSION TITLE
Trace id generation for logs

### DISCUSSION THREAD

**Discussion by DGolubets** [apollographql/router#4950] [Q&A]
Hi,

Can the router generate trace id if it's not provided in the header?
Just to link different logs together.

**Comment by bnjjj** [comment]
Yes and you can choose to display trace_id in your logs if you want https://www.apollographql.com/docs/router/configuration/telemetry/exporters/logging/stdout cf `display_trace_id`

**Reply by DGolubets** [reply]
My experience is that if I just enable this option I don't see trace_id in my logs unless I explicitly configure and pass trace id in request headers.
